src/backend/pluginLoaders.js

```diff
@@ -82,3 +95,31 @@ export async function updateMirror(
   await Promise.all(tasks);
   return {project, cache};
 }
+
+/**
+ * Creates PluginGraphs containing all plugins requested by the Project.
+ */
+export async function createPluginGraphs(
+  {github, discourse}: PluginLoaders,
+  {githubToken}: GraphEnv,
+  {cache, project}: CachedProject
+): Promise<PluginGraphs> {
+  const tasks: Promise<WeightedGraphT>[] = [];
+  if (project.discourseServer) {
+    tasks.push(discourse.createGraph(project.discourseServer, cache));
+  }
+  if (project.repoIds.length) {
+    if (!githubToken) {
```

Comment:
We shouldn't need this check now -- we already have the cachedProject.

Reply:
Flow disagrees :sweat_smile:, the `GraphEnv` accepts the same `?GithubToken` as elsewhere, meaning we need to refine it if we want to use it as a `GithubToken`.

There's a good reason it should be optional here. Such as loading a Project without any Github repos. Whether it's a CachedProject or not, it might not specify Github repos, thus not needing a token. Which is why the check happens after
` if (project.repoIds.length)`.

Reply:
Though, we can remove this token entirely when we've fixed this issue #1580

Reply:
Yeah, I meant that I wanted it to get removed per #1580. Fine with that being a different PR though.

Reply:
There's some complexity involved that's blocking #1580, so I think that would be good as it's own PR.